src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -108,6 +93,21 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
             process.BeginOutputReadLine();
             process.BeginErrorReadLine();
             processSpec.OnStart?.Invoke(process.Id);
+
+            process.WaitForExitAsync().ContinueWith(t =>
+            {
+                startupComplete.Wait();
+
+                if (processSpec.ThrowOnNonZeroReturnCode && process.ExitCode != 0)
+                {
+                    processLifetimeTcs.TrySetException(new InvalidOperationException(
+                        $"Command {processSpec.ExecutablePath} {processSpec.Arguments} returned non-zero exit code {process.ExitCode}"));
+                }
+                else
+                {
+                    processLifetimeTcs.TrySetResult(new ProcessResult(process.ExitCode));
+                }
+            }, TaskScheduler.Current);
```

Comment:
```suggestion
            }, TaskScheduler.Default);
```

This should always use the thread pool, not some a custom TaskScheduler, if there is one.

Reply:
Switched to `TaskScheduler.Default`. Omitted the continuation options because, like you said, shouldn't make a tangible difference.